### PR TITLE
Code simplifications in AbstractMojo

### PR DESCRIPTION
**Pull request by rhowe**
Clean up a few overly-verbose code constructs:

- Replace an unnecessary use of `StringBuilder` with string concatenation.
- Remove some code duplication when selecting which archiver to use.
- Eliminate an `else` branch by returning early in `AbstractMojo#projectHasAlreadySetAnArtifact()`
- Remove an unnecessary local variable in `AbstractMojo#hasClassifier()`

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java [hunk 1]
```diff
@@ -196,16 +196,17 @@ protected File getJarFile( File basedir, String resultFinalName, String classifi
             throw new IllegalArgumentException( "finalName is not allowed to be null" );
         }
 
-        StringBuilder fileName = new StringBuilder( resultFinalName );
-
+        final String fileName;
```
Comment (olamy) [inline src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java:199]: not sure what is the need if final here?

File: src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java [hunk 2]
```diff
@@ -243,18 +244,11 @@ public File createArchive()
             }
         }
 
+        final String archiverName = containsModuleDescriptor ? "mjar" : "jar";
```
Comment (olamy) [inline src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java:247]: why final? no real need of this

**Review by olamy — CHANGES_REQUESTED** [review]
lgtm
just minor comment.
maybe it's only me but I find adding final not adding any real value here

File: src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java (hunk 2 above)
Comment (michael-o) [reply src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java:247]: oh oh, I know people they insinst on final...

File: src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java (hunk 1 above)
Comment (rhowe) [reply src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java:199]: No real need, it's mostly just syntactic sugar in this case. It's somewhat a statement of intent indicating that the value will be calculated only once.
This was the kind of feedback I was seeking though - if you're happy with the changeset in principle, I'll take the `final`s out, cut a Jira ticket and make the PR more presentable etc.

Comment (olamy) [reply src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java:199]: all good. please remove those `final` as I don't think they make sense for a local variable. except adding more useless code.
btw I'm more a sugar free person if it's not really needed 🤣 
if you want to create a jira why not but I'm not sure we need a Jira for such code polishing.
it's up2u

Comment (rhowe) [reply src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java]: The less I need to touch Jira the better. OK, I've removed the `final`s and updated the PR description

Comment (olamy) [reply src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java]: @rhowe great. Thanks! Still on final to remove  ;)

Comment (rhowe) [reply src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java]: Oops, OK the final `final` is gone

**Pull request merged**